### PR TITLE
Generalize upgrade scripts to test repeat upgrades

### PR DESCRIPTION
**Pull request by mrice32**
To run a normal upgrade and verify it, you should follow the following steps:
```
$(npm bin)/truffle exec --network mainnet-fork ./scripts/umip-3/1_Propose.js
$(npm bin)/truffle exec --network mainnet-fork ./scripts/umip-3/2_VoteSimulate.js
$(npm bin)/apply-registry
 $(npm bin)/truffle exec --network mainnet-fork ./scripts/umip-3/3_Verify.js
```

To run another upgrade on top of the previous one:
```
$(npm bin)/truffle exec --network mainnet-fork ./scripts/umip-3/1_Propose.js
$(npm bin)/truffle exec --network mainnet-fork ./scripts/umip-3/2_VoteSimulate.js --repeated
$(npm bin)/apply-registry
 $(npm bin)/truffle exec --network mainnet-fork ./scripts/umip-3/3_Verify.js
```

This also adds a second flag to `2_VoteSimulate.js` to revert changes after a run: `--revert`. Previously, reverting was the default behavior.

### CODE REVIEW

File: core/scripts/umip-3/3_Verify.js [hunk 1]
```diff
@@ -16,13 +16,13 @@ const ownerRole = "0";
...
 // New addresses of ecosystem components after porting from `Propose.js`
 const upgradeAddresses = {
-  Voting: "0x7492cdbc126ffc05c32249a470982173870e95b0",
-  Registry: "0x46209e15a14f602897e6d72da858a6ad806403f1",
-  Store: "0x74d367e2207e52f05963479e8395cf44909f075b",
-  FinancialContractsAdmin: "0x3b99859be43d543960803c09a0247106e82e74ee",
-  IdentifierWhitelist: "0x9e39424eab9161cc3399d886b1428cba71586cb8",
-  Governor: "0x878cfedb234c226ddefd33657937af74c17628bf",
-  Finder: "0x40f941E48A552bF496B154Af6bf55725f18D77c3" // Finder was no upgraded in UMIP3
+  Voting: Voting.address,
```
Comment (chrismaree) [inline core/scripts/umip-3/3_Verify.js:19]: how exactly will these have the latest addresses from script 1? this is due to the apply-registry I'm guessing?

Comment (mrice32) [reply core/scripts/umip-3/3_Verify.js:19]: Yep, you have to apply registry after 2, but before 3.

Comment (chrismaree) [reply core/scripts/umip-3/3_Verify.js:19]: great

File: core/scripts/umip-3/3_Verify.js [hunk 2]
```diff
@@ -16,13 +16,13 @@ const ownerRole = "0";
...
+  Registry: Registry.address,
+  Store: Store.address,
+  FinancialContractsAdmin: FinancialContractsAdmin.address,
+  IdentifierWhitelist: IdentifierWhitelist.address,
+  Governor: Governor.address,
+  Finder: Finder.address // Finder was no upgraded in UMIP3
```
Comment (ptare) [inline core/scripts/umip-3/3_Verify.js:25]: s/no/not

**Pull request merged**